core/src/main/java/org/sonarsource/sonarlint/core/container/connected/update/PluginReferencesDownloader.java

```diff
@@ -82,9 +81,11 @@ private static boolean checkVersion(@Nullable String version, @Nullable String m
     return true;
   }
 
-  public PluginReferences fetchPluginsTo(Path dest, List<SonarAnalyzer> analyzers) {
+  public PluginReferences fetchPluginsTo(Path dest, List<SonarAnalyzer> analyzers, ProgressWrapper progress) {
     PluginReferences refs = fetchPlugins(analyzers);
+    int i = 0;
     for (PluginReference ref : refs.getReferenceList()) {
+      progress.setProgressAndCheckCancel("Loading plugin " + ref.getKey(), i / (float) refs.getReferenceList().size());
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Use a "double" or "BigDecimal" instead. ([squid:S2164](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2164))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarlint.core%3Asonarlint-core-parent&pullRequest=197&issues=AWOCPMWSLjbbHpSefgss&open=AWOCPMWSLjbbHpSefgss)

[ISSUE_KEY]: AWOCPMWSLjbbHpSefgss